src/EditorFeatures/Core/Shared/Tagging/Utilities/TagSpanIntervalTree.cs

```diff
@@ -92,8 +79,21 @@ private void AppendIntersectingSpansInSortedOrder(SnapshotSpan snapshotSpan, Seg
             result.Add(GetTranslatedTagSpan(tagSpan, snapshot, _spanTrackingMode));
     }
 
-    public IEnumerable<TagSpan<TTag>> GetSpans(ITextSnapshot snapshot)
-        => _tree.Select(tn => GetTranslatedITagSpan(tn, snapshot));
+    /// <summary>
+    /// Gets all the tag spans in this tree, remapped to <paramref name="snapshot"/>, and returns them as a <see
+    /// cref="NormalizedSnapshotSpanCollection"/>.
+    /// </summary>
+    public NormalizedSnapshotSpanCollection GetSnapshotSpanCollection(ITextSnapshot snapshot)
```

Comment:
several calls to GetSpans were replaced to a call to this.  this version returns the right data they needed, while also pooling internal values needed to compute it.